Tests/tests/test_SystemServicesTest.cpp

```diff
@@ -0,0 +1,324 @@
+ * following copyright and licenses apply:
+ *
+ * Copyright 2022 RDK Management
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <gtest/gtest.h>
+#include "SystemServices.h"
+#include "IarmBusMock.h"
+#include "RfcApiMock.h"
+#include "WrapsMock.h"
+using namespace WPEFramework;
+using namespace std;
+
+namespace {
+const string iarmName = _T("Thunder_Plugins");
+}
+
+class SystemServicesTest : public::testing::Test
+{
+    protected:
+    IarmBusImplMock iarmBusImplMock;
+    RfcApiImplMock rfcApiImplMock;
+    WrapsImplMock wrapsImplMock;
+    Core::ProxyType<Plugin::SystemServices> systemplugin;
+    Core::JSONRPC::Connection connection;
+    Core::JSONRPC::Handler *handler;
+    Core::JSONRPC::Handler *handlerV2;
+    string response;
+ 
+public:
+    SystemServicesTest()
+    :connection(1,0)
+    {
+	    IarmBus::getInstance().impl = &iarmBusImplMock;
+        RfcApi::getInstance().impl = &rfcApiImplMock;
+        Wraps::getInstance().impl = &wrapsImplMock;
+    }
+   
+    virtual void SetUp()
+    {
+    }
+
+    virtual void TearDown()
+    {
+    }
+
+    ~SystemServicesTest()
+    {
+        IarmBus::getInstance().impl = nullptr;
+        RfcApi::getInstance().impl = nullptr;
+        Wraps::getInstance().impl = nullptr;	
+    }
+};
+
+TEST_F(SystemServicesTest, RegisterMethods)
+{
+    EXPECT_CALL(iarmBusImplMock, IARM_Bus_Call)
+        .Times(1)
+        .WillOnce(
+                [](const char* ownerName, const char* methodName, void* arg, size_t argLen) {
+                    EXPECT_TRUE(strcmp(methodName, "DaemonSysModeChange") == 0);
+                    return IARM_RESULT_SUCCESS;
+                });
+
+    EXPECT_CALL(wrapsImplMock, system(::testing::_))
+        .Times(1)
+        .WillOnce(::testing::Invoke(
+            [&](const char* command) {
+                EXPECT_EQ(string(command), string(_T("rm -f /opt/warehouse_mode_active")));
+
+                return 0;
+            }));
+
+	    EXPECT_CALL(rfcApiImplMock, getRFCParameter(::testing::_, ::testing::_, ::testing::_))
+        .Times(1)
+        .WillOnce(::testing::Invoke(
+            [](char *pcCallerID, const char* pcParameterName, RFC_ParamData_t* pstParamData) {
+                EXPECT_EQ(strcmp(pcParameterName, "Device.DeviceInfo.X_RDKCENTRAL-COM_RFC.Feature.Power.PwrMgr2.Enable"), 0);
+                return WDMP_SUCCESS;
+        }));
+
+        systemplugin = Core::ProxyType<Plugin::SystemServices>::Create();
```

Comment:
many test cases create plugin instance. please make another test fixture SystemServicesPluginTest that derives from SystemServicesTest and creates plugin, and handlers. base class SystemServicesTest constructor would be called before SystemServicesPluginTest and init the mocks before plugin

Reply:
Updated